app/services/reject_npq_application.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+class RejectNPQApplication
+  include ActiveModel::Model
+  include ActiveModel::Attributes
+  include ActiveModel::Validations::Callbacks
+
+  attribute :npq_application
+
+  validates :npq_application, presence: { message: I18n.t(:missing_npq_application) }
+  validate :not_already_rejected
+  validate :cannot_change_from_accepted
+
+  def call
+    return self unless valid?
+
+    npq_application.update!(lead_provider_approval_status: "rejected")
+    npq_application
+  end
+
+private
+
+  def not_already_rejected
+    return if npq_application.blank?
+
+    errors.add(:npq_application, I18n.t("activerecord.errors.models.npq_application.attributes.lead_provider_approval_status.has_already_been_rejected")) if npq_application.rejected?
```

Comment:
same question here around error message, let me know your thoughts

Reply:
Hi @cwrw tks for your review. Addressed in https://github.com/DFE-Digital/early-careers-framework/pull/2674/commits/8c9fde545669f063b8a66cd2c524248a8025fdb5